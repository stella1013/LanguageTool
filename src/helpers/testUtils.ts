//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/