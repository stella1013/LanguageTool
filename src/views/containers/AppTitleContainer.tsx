import React from 'react';

class AppTitleContainer extends React.Component<any, {}> {
	render() {
		return <div data-test="component-apptitle"><h2>{this.props.title}</h2></div>;
	}
}

export default AppTitleContainer;
