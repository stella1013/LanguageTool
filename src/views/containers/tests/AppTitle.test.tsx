import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { findByTestAttr } from '../../../helpers/testUtils';
import thunk from 'redux-thunk';
import AppTitleContainer from '../AppTitleContainer';



describe('AppTitle', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */
    const createMockStore = configureStore([thunk]);
    const initialState = {};
    const store = createMockStore(initialState);
    let wrapper: ShallowWrapper;
	
	describe('rendering', () => {
       //let wrapper: any;
		beforeEach(() => {
            wrapper = shallow(<AppTitleContainer title="title"/>);
		});

		it('renders without crashing', () => {
			const component = findByTestAttr(wrapper, 'component-apptitle');
			expect(component.length).toBe(1);
		});

		it('render matches snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});

		
	});

	describe('state updates', () => {});
});
