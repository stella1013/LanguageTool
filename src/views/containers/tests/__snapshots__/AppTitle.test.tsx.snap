// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle rendering render matches snapshot 1`] = `ShallowWrapper {}`;
